framework/tlvf/src/src/BaseClass.cpp

```diff
@@ -41,3 +43,22 @@ bool BaseClass::buffPtrIncrementSafe(size_t length)
     m_buff_ptr__ += length;
     return true;
 }
+
+bool BaseClass::finalize()
+{
+    if (m_parse__) {
+        TLVF_LOG(ERROR) << "finalize() called but m_parse__ is set";
+        return false;
+    }
+    if (m_finalized) {
+        TLVF_LOG(DEBUG) << "finalize() called for already finalized class";
+        return true;
+    }
+    if (!isPostInitSucceeded()) {
+        TLVF_LOG(ERROR) << "post init check failed";
+        return false;
+    }
+    class_swap();
+    m_finalized = true;
```

Comment:
Hm, this is risky...

Before, the m_swapped field of ClassList was checked in finalize and it was not done again if it was already swapped. Now, only the finalized flag is checked. Thus, the following sequence would have worked previously but no longer works: `cmdu.swap(); cmdu.finalize();` And there are probably more corner cases.

I think it would be good to do a review of all explicit cmdu.swap and foo.class_swap calls and explain why they're OK (for example, in a reply to this comment). And of course testing.

Reply:
Or make swap protected and never use it outside of the class